src/AttestationData.ts

```diff
@@ -0,0 +1,408 @@
+type TContextField = string | {type: string; data: string}
+
+///////////////////////////////////////////////////
+// Base attestation dataStr type
+///////////////////////////////////////////////////
+export interface IBaseAtt {
+  '@context': TContextField | Array<TContextField>
+  // Date/time attestation was performed
+  date?: string
+
+  // Secondary provider
+  provider?: string
+
+  // Unique ID for attestation subject from provider
+  id?: number | string
+
+  // Date/time when attestation should be considered void (e.g., for credential expiry)
+  expiry_date?: string
+
+  // Optional summary object (often useful for multiple-account types - otherwise, the data field is preferable for most of these fields, for simplicity's sake
+  summary?: {
+    // Single date/time during which attestation was applicable
+    date?: TDate
```

Comment:
Will it ever be `date` and `start_date`+`end_date` or just `date` OR `start_date`+`end_date`.

If it's the latter maybe this type should be:

```
type TBaseSummary = {
  generality?: number
}

type TSummaryWithDate = TBaseSummary & {
  date: TDate
}

type TSummaryWithStartEndDate = TBaseSummary & {
  start_date: TDate
  end_date: TDate
}

type TSummary = TSummaryWithDate | TSummaryWithStartEndDate
```

Reply:
That's at the base type for all attestations, so it could certainly be one of start_date or end_date without the other.